client/internal/receivedprocessor.go

```diff
@@ -121,124 +146,15 @@ func (r *receivedProcessor) rcvFlags(
-
-	// Get the hash of the status before we update it.
-	prevStatus := r.clientSyncedState.RemoteConfigStatus()
-	var prevCfgStatusHash []byte
-	if prevStatus != nil {
-		prevCfgStatusHash = prevStatus.Hash
-	}
-
-	// Remember the status for the future use.
-	_ = r.clientSyncedState.SetRemoteConfigStatus(cfgStatus)
-
-	// Check if the status has changed by comparing the hashes.
-	if !bytes.Equal(prevCfgStatusHash, cfgStatus.Hash) {
-		// Let the Agent know about the status.
-		r.callbacks.SaveRemoteConfigStatus(ctx, cfgStatus)
-
-		// Include the config status in the next message to the Server.
-		r.sender.NextMessage().Update(func(msg *protobufs.AgentToServer) {
-			msg.RemoteConfigStatus = cfgStatus
-		})
-
-		reportStatus = true
-	}
-
-	// Report the effective configuration if it changed and remote config was applied
-	// successfully.
-	if changed && applyErr == nil {
-		r.sender.NextMessage().Update(func(msg *protobufs.AgentToServer) {
-			msg.EffectiveConfig = effective
-		})
-		reportStatus = true
-	}
-	return reportStatus
-}
-
-func (r *receivedProcessor) rcvConnectionSettings(ctx context.Context, settings *protobufs.ConnectionSettingsOffers) {
-	if settings == nil {
+func (r *receivedProcessor) rcvOpampConnectionSettings(ctx context.Context, settings *protobufs.ConnectionSettingsOffers) {
+	if settings == nil || settings.Opamp == nil {
 		return
 	}
 
-	if settings.Opamp != nil {
-		err := r.callbacks.OnOpampConnectionSettings(ctx, settings.Opamp)
-		if err == nil {
-			// TODO: verify connection using new settings.
-			r.callbacks.OnOpampConnectionSettingsAccepted(settings.Opamp)
-		}
-	}
-
-	r.rcvOwnTelemetryConnectionSettings(
-		ctx,
-		settings.OwnMetrics,
-		types.OwnMetrics,
-		r.callbacks.OnOwnTelemetryConnectionSettings,
-	)
-
-	r.rcvOwnTelemetryConnectionSettings(
-		ctx,
-		settings.OwnTraces,
-		types.OwnTraces,
-		r.callbacks.OnOwnTelemetryConnectionSettings,
-	)
-
-	r.rcvOwnTelemetryConnectionSettings(
-		ctx,
-		settings.OwnLogs,
-		types.OwnLogs,
-		r.callbacks.OnOwnTelemetryConnectionSettings,
-	)
-
-	for name, s := range settings.OtherConnections {
-		r.rcvOtherConnectionSettings(ctx, s, name, r.callbacks.OnOtherConnectionSettings)
-	}
-}
-
-func (r *receivedProcessor) rcvOwnTelemetryConnectionSettings(
-	ctx context.Context,
-	settings *protobufs.TelemetryConnectionSettings,
-	telemetryType types.OwnTelemetryType,
-	callback func(ctx context.Context, telemetryType types.OwnTelemetryType, settings *protobufs.TelemetryConnectionSettings) error,
-) {
-	if settings != nil {
-		callback(ctx, telemetryType, settings)
-	}
-}
-
-func (r *receivedProcessor) rcvOtherConnectionSettings(
-	ctx context.Context,
-	settings *protobufs.OtherConnectionSettings,
-	name string,
-	callback func(ctx context.Context, name string, settings *protobufs.OtherConnectionSettings) error,
-) {
-	if settings != nil {
-		callback(ctx, name, settings)
+	err := r.callbacks.OnOpampConnectionSettings(ctx, settings.Opamp)
```

Comment:
It is a multi-step process:
1. Client receives OpAMP connection settings.
2. OnOpampConnectionSettings is called. Agent can verify the offer and accept or reject (e.g. check certificate, etc).
3. If (2) is successful the Client connects using the new settings to verify the actually work.
4. If (3) is successful OnOpampConnectionSettingsAccepted is called so that the Agent can persist the settings.

We don't have this in the example, and for step 3 we have a TODO.